### PR TITLE
Add factory with a given uuid

### PR DESCRIPTION
**Pull request by ouaibsky**

### CODE REVIEW

**Comment by jvalkeal** [comment]
Would you mind singing our CLA https://cla.pivotal.io/

Thx

**Comment by ouaibsky** [comment]
Weird, I'll already do it, I'm gonna retry

**Comment by jvalkeal** [comment]
ah ok, it's prob fine then. I haven't yet setup automatic check for PR's and assumed you didn't sign it as this new CLA site is relatively new.

**Comment by ouaibsky** [comment]
Is there is any automatic build ?

**Comment by jvalkeal** [comment]
nope, we don't have any builds for PR's as there's been no need in a past. I may try to add travis and see if its open source version gives enough resources to run  tests. Anyway, starting to look these PR's now.

File: spring-statemachine-core/src/main/java/org/springframework/statemachine/config/AbstractStateMachineFactory.java [hunk 1]
```diff
@@ -37,55 +28,32 @@
...
-import org.springframework.statemachine.config.model.ChoiceData;
-import org.springframework.statemachine.config.model.EntryData;
-import org.springframework.statemachine.config.model.ExitData;
-import org.springframework.statemachine.config.model.HistoryData;
-import org.springframework.statemachine.config.model.JunctionData;
-import org.springframework.statemachine.config.model.StateData;
-import org.springframework.statemachine.config.model.StateMachineModel;
-import org.springframework.statemachine.config.model.TransitionData;
-import org.springframework.statemachine.config.model.TransitionsData;
+import org.springframework.statemachine.config.model.*;
```
Comment (jvalkeal) [inline spring-statemachine-core/src/main/java/org/springframework/statemachine/config/AbstractStateMachineFactory.java]: Don't add import wildcards(if you're using idea then I think it uses that automatically after some amount of imports from same package.).

File: spring-statemachine-core/src/main/java/org/springframework/statemachine/ObjectStateMachine.java [hunk 2]
```diff
@@ -57,8 +58,8 @@ public ObjectStateMachine(Collection<State<S, E>> states, Collection<Transition<
 	 */
 	public ObjectStateMachine(Collection<State<S, E>> states, Collection<Transition<S, E>> transitions,
 			State<S, E> initialState, Transition<S, E> initialTransition,
-			Message<E> initialEvent, ExtendedState extendedState) {
-		super(states, transitions, initialState, initialTransition, initialEvent, extendedState);
+			Message<E> initialEvent, ExtendedState extendedState, UUID uuid) {
+		super(states, transitions, initialState, initialTransition, initialEvent, extendedState, uuid);
```
Comment (jvalkeal) [inline spring-statemachine-core/src/main/java/org/springframework/statemachine/ObjectStateMachine.java]: javadocs for uuid
```
ObjectStateMachine.java:59: warning: no @param for uuid
	public ObjectStateMachine(Collection<State<S, E>> states, Collection<Transition<S, E>> transitions,
```

File: spring-statemachine-core/src/main/java/org/springframework/statemachine/support/AbstractStateMachine.java [hunk 3]
```diff
@@ -134,7 +121,7 @@ public AbstractStateMachine(Collection<State<S, E>> states, Collection<Transitio
 	 */
 	public AbstractStateMachine(Collection<State<S, E>> states, Collection<Transition<S, E>> transitions,
 			State<S, E> initialState, ExtendedState extendedState) {
-		this(states, transitions, initialState, null, null, extendedState);
+		this(states, transitions, initialState, null, null, extendedState, null);
```
Comment (jvalkeal) [inline spring-statemachine-core/src/main/java/org/springframework/statemachine/support/AbstractStateMachine.java]: javadocs for uuid
```
AbstractStateMachine.java:137: warning: no @param for uuid
	public AbstractStateMachine(Collection<State<S, E>> states, Collection<Transition<S, E>> transitions,
```

**Comment by jvalkeal** [comment]
Doesn't build as test compile fails:
```
$ ./gradlew clean build
...
:spring-statemachine-core:compileTestJava/home/jvalkealahti/repos/tmp/ssm/spring-statemachine/spring-statemachine-core/src/test/java/org/springframework/statemachine/RegionMachineTests.java:109: error: no suitable constructor found for ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,<null>,<null>)
		ObjectStateMachine<TestStates, TestEvents> machine = new ObjectStateMachine<TestStates, TestEvents>(states, transitions, stateSI, initialTransition, null, null);
		                                                     ^
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>) is not applicable
      (actual and formal argument lists differ in length)
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,Message<TestEvents>,ExtendedState,UUID) is not applicable
      (actual and formal argument lists differ in length)
/home/jvalkealahti/repos/tmp/ssm/spring-statemachine/spring-statemachine-core/src/test/java/org/springframework/statemachine/RegionMachineTests.java:175: error: no suitable constructor found for ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,<null>,<null>)
		ObjectStateMachine<TestStates, TestEvents> machine11 = new ObjectStateMachine<TestStates, TestEvents>(states11, transitions11, stateS111, initialTransition11, null, null);
		                                                       ^
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>) is not applicable
      (actual and formal argument lists differ in length)
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,Message<TestEvents>,ExtendedState,UUID) is not applicable
      (actual and formal argument lists differ in length)
/home/jvalkealahti/repos/tmp/ssm/spring-statemachine/spring-statemachine-core/src/test/java/org/springframework/statemachine/RegionMachineTests.java:188: error: no suitable constructor found for ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,<null>,<null>)
		ObjectStateMachine<TestStates, TestEvents> machine12 = new ObjectStateMachine<TestStates, TestEvents>(states12, transitions12, stateS121, initialTransition12, null, null);
		                                                       ^
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>) is not applicable
      (actual and formal argument lists differ in length)
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,Message<TestEvents>,ExtendedState,UUID) is not applicable
      (actual and formal argument lists differ in length)
/home/jvalkealahti/repos/tmp/ssm/spring-statemachine/spring-statemachine-core/src/test/java/org/springframework/statemachine/RegionMachineTests.java:205: error: no suitable constructor found for ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,RegionState<TestStates,TestEvents>,Transition<TestStates,TestEvents>,<null>,<null>)
		ObjectStateMachine<TestStates, TestEvents> machine = new ObjectStateMachine<TestStates, TestEvents>(states, transitions, stateR, initialTransition, null, null);
		                                                     ^
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>) is not applicable
      (actual and formal argument lists differ in length)
    constructor ObjectStateMachine.ObjectStateMachine(Collection<State<TestStates,TestEvents>>,Collection<Transition<TestStates,TestEvents>>,State<TestStates,TestEvents>,Transition<TestStates,TestEvents>,Message<TestEvents>,ExtendedState,UUID) is not applicable
      (actual and formal argument lists differ in length)
4 errors
 FAILED

FAILURE: Build failed with an exception.
```

File: spring-statemachine-core/src/main/java/org/springframework/statemachine/config/AbstractStateMachineFactory.java (hunk 1 above)
Comment (ouaibsky) [inline spring-statemachine-core/src/main/java/org/springframework/statemachine/config/AbstractStateMachineFactory.java]: ya it's five and no comment on this on dev guidelines

**Comment by jvalkeal** [comment]
@ouaibsky right, would you mind fixing conflicts and rebasing atop of latest master. Your PR I merged broke this PR.

**Comment by ouaibsky** [comment]
rebase done

**Comment by jvalkeal** [comment]
ok, I'm back from pto. Think this looks good for now, merging.

**Comment by jvalkeal** [comment]
squashed/merged per https://github.com/spring-projects/spring-statemachine/commit/e6f10b60f55a85482f46e55f7801636c680b7859

**Pull request closed without merging**